src/Microsoft.IdentityModel.Clients.ActiveDirectory/Internal/Http/AdalHttpClient.cs

```diff
@@ -128,8 +128,7 @@ public AdalHttpClient(string uri, CallState callState)
                     if (ex.WebResponse.StatusCode == HttpStatusCode.BadRequest &&
                         tokenResponse.Error == AdalError.InteractionRequired)
                     {
-                        throw new AdalClaimChallengeException(tokenResponse.Error, tokenResponse.ErrorDescription,
-                            tokenResponse.Claims);
+                        throw new AdalClaimChallengeException(tokenResponse.Error, tokenResponse.ErrorDescription, ex.InnerException, tokenResponse.Claims);
```

Comment:
passed in exception seems to be incorrect. check what exception object is passed in when it is not claims challenge exception

Reply:
Done.